src/server/checker/CheckerService.ts

```diff
@@ -92,6 +100,10 @@ export class CheckerService {
   ) => Promise<GetPublishedCheckerWithoutDraftCheckerDTO | null> = async (
     id
   ) => {
+    // Check if checker is active
+    const checker = await this.CheckerModel.findByPk(id, {
```

Comment:
Given that now we allow multiple users to access/modify the same checker, we should wrap operations consecutive SQL operations up in a transaction to ensure a consistent results. In this case, for the initial publishing of a checker, there might be a situation where you return `isActive` is `false` but also yet return a published checker.